### PR TITLE
Add `FirstPartyModeToggle` component

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #9649 

## Relevant technical choices

This PR adds the first-party mode toggle component.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/ads/components/settings/SettingsForm.js [hunk 1]
```diff
@@ -145,6 +147,7 @@ export default function SettingsForm() {
 							}
 						) }
 					</ConversionTrackingToggle>
+					{ fpmEnabled && <FirstPartyModeToggle /> }
```
Comment (techanvil) [inline assets/js/modules/ads/components/settings/SettingsForm.js:150]: Rendering the toggle in the settings edit view is actually out of scope for this issue, it's due in https://github.com/google/site-kit-wp/issues/9655.

We _can_ keep it here, but I'd suggest removing it so it can be implemented and QA'd as part of the above issue.

File: assets/js/modules/analytics-4/components/settings/SettingsForm.js [hunk 2]
```diff
@@ -90,6 +94,7 @@ export default function SettingsForm( { hasModuleAccess } ) {
 						}
 					) }
 				</ConversionTrackingToggle>
+				{ fpmEnabled && <FirstPartyModeToggle /> }
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/settings/SettingsForm.js:97]: As above, this is out of scope for this issue.

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @nfmohit, this is generally looking great.

I've left a few trivial comments and a couple of more significant ones, please take a look.

File: assets/js/components/first-party-mode/FirstPartyModeToggle.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	}, [ isFirstPartyModeEnabled, setFirstPartyModeEnabled ] );
+
+	return (
+		<div
+			className={ classnames(
+				'googlesitekit-first-party-mode-toggle',
+				className
+			) }
+		>
+			{ loading && <ProgressBar small /> }
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply assets/js/components/first-party-mode/FirstPartyModeToggle.js]: Updated, please let me know if this looks good now, thank you!

File: assets/js/components/first-party-mode/FirstPartyModeToggle.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import SubtleNotification from '../notifications/SubtleNotification';
+
+export default function FirstPartyModeToggle( { className } ) {
+	const isFirstPartyModeEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).isFirstPartyModeEnabled()
+	);
+	const loading = useSelect( ( select ) =>
+		select( CORE_SITE ).isFetchingGetFPMServerRequirementStatus()
+	);
+	const metServerRequirements = useSelect( ( select ) => {
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply assets/js/components/first-party-mode/FirstPartyModeToggle.js]: Updated, thank you!

File: assets/js/components/first-party-mode/FirstPartyModeToggle.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+		return fpmHealthy && scriptAccessEnabled;
+	} );
+
+	const { fetchGetFPMServerRequirementStatus, setFirstPartyModeEnabled } =
+		useDispatch( CORE_SITE );
+
+	// Fetch the server requirement status on mount.
+	useMount( () => {
+		fetchGetFPMServerRequirementStatus();
+	} );
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply assets/js/components/first-party-mode/FirstPartyModeToggle.js]: Updated, thank you!

File: assets/js/modules/ads/components/settings/SettingsForm.js (hunk 1 above)
Comment (nfmohit) [reply assets/js/modules/ads/components/settings/SettingsForm.js]: Removed, thank you!

**Comment by nfmohit** [comment]
> Thanks @nfmohit, this is generally looking great.
> 
> I've left a few trivial comments and a couple of more significant ones, please take a look.

Thank you for the kind review, @techanvil ! I've updated the PR accordingly. Please let me know if this needs any other changes.

File: assets/js/components/first-party-mode/FirstPartyModeToggle.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/components/first-party-mode/FirstPartyModeToggle.js]: Thanks Nahid, that's great :tada:

**Review by techanvil — APPROVED** [review]
Thanks @nfmohit, that's great. LGTM!
